internal/common/gamestate/clientgamestate.go

```diff
@@ -0,0 +1,13 @@
+package gamestate
+
+// ClientGameState contains game state only for a specific client.
+type ClientGameState struct {

```

Comment:
Good shout making a separate ClientGameState struct. So we do still have the GameState with all the ClientInfos for server funcs?

Reply:
yes :)

Reply:
so when we update an islands individual resource pool, do we have to update it in both places?

Reply:
nope, you only update it in the server's own `gameState`. When you call `gameStateUpdate`, it copies it over to the corresponding `ClientGameState` :)

Reply:
More info:

You get the corresponding `ClientGameState` by doing for example for team 1:
```
s.gameState.GetClientGameStateCopy(shared.Team1)
```
This gets a copy based on `s.gameState`.

Reply:
oh sweet!